### PR TITLE
Add node / nvm build error note to the README.md

### LINKED ISSUE #91: Update CONTRIBUTING.md to include nvm / node install instructions
Need to document the fact the the distro packaged versions of node throw build errors on linux and we need to use node releases downloaded via the node version manager instead.

### PR DESCRIPTION
**Pull request by Tweekism**
Close #91

### CODE REVIEW

**Comment by Tweekism** [comment]
image

**Comment by Tweekism** [comment]
I've put it here for now, with the build instructions.

If you want more details it might be better on its own page, like a troubleshooting or on the contributing page. To keep the front page README.md from getting too over whelming.

**Comment by jannis-baum** [comment]
> I've put it here for now, with the build instructions.
> 
> If you want more details it might be better on its own page, like a troubleshooting or on the contributing page. To keep the front page README.md from getting too over whelming.

Yes sounds good. We can just link to the contributing under "compile yourself" on the readme and then put the instructions there. Would be great if you could also figure out the minimal version of Node required for SEA and add it there as well :)

**Comment by Tweekism** [comment]
Thats a good idea, then we can include more details, like we could list minimum versions for all the dependencies if we so wished.

I'll do that now.

**Comment by jannis-baum** [comment]
Hey, just as a heads-up: I just added Markdown linting to the repo, please rebase to main and make sure your changes will pass the linting rules :)

**Comment by Tweekism** [comment]
Hi guys, 

Getting back to this today...

Is jq no longer a requirement at all?

**Comment by jannis-baum** [comment]
Awesome!

Yes, `jq` is no longer a dependency :) We are dependency-free at runtime now.

`zip` still is a build dependency. We considered switching to `tar` but it got a bit ugly due to differences between BSD `tar` (on macOS) and GNU `tar` (on Linux) so we ended up on the thought that it's not quite worth it, especially since it is not needed at runtime anyways

**Comment by Tweekism** [comment]
Cool.

Question for both of you @jannis-baum @tuurep, when you are doing ./configure <install_path> ...Where are you guys installing to?

I'm installing to ~/.local/bin/ because I have write access there (without needing root) and it is on $path by default, but that might be a Fedora thing, doesn't seem to be the case on Ubuntu.

**Comment by jannis-baum** [comment]
We had a discussion about that topic starting in https://github.com/jannis-baum/Vivify/pull/68#issuecomment-2212560005 and the next few comments.

**tldr;** I think Tuure installs to `~/.local/bin`. On macOS we don't have any user-writable directory on `$PATH` by default so I started my own convention a long time ago at `~/.bin` which I should probably change to be the conventional `~/.local/bin` ^^ Installing to something you have write permissions to seems like the most sensible approach to us :)

**Comment by jannis-baum** [comment]
Hey this is super out of context but just as a heads-up for both of you @tuurep @Tweekism: I edited your PR descriptions to explicitly say `Close #<issue-number>` because this makes GitHub understand that the issue is linked to the PR and will also auto-close it on merge.

Maybe this is something we should also mention in the contribution guide? @Tweekism could you add it on your PR here since you are already editing that file anyways?☺️

**Comment by Tweekism** [comment]
Sure can mate, I was wondering why I wasn't getting the little icon on the issues page, I thought I had added it.

**Comment by Tweekism** [comment]
Hey peeps,

This is ready for you to have a look at, sorry it took so long, since it is a lot of change I know I'm holding up anyone else who wants to update the docs. Is it too much?

Note there is one "TODO:" in there, and that is the command to install the required dependencies on MacOS. Could you let me know? Or do we not want to go to that level of detail. I kind of modeled it after the Ladybird browser project that I flirted with a little bit 😅

**Comment by tuurep** [comment]
Yes I install to `~/.local/bin` by the way

**Comment by Tweekism** [comment]
Hey @tuurep,

Yeah I went back and had a look at that thread. Is that a default path on Arch?

**Comment by Tweekism** [comment]
> > doesn't seem to be the case on Ubuntu.
> 
> Oh.. Interesting lol

To be fair, I only checked on Ubuntu server, I’ll have a look on Ubuntu desktop see if it’s any different.

**Comment by tuurep** [comment]
Actually I went to double check if `~/.local/bin` is on `$PATH` by default on Arch, and **it's not** :smile:

I have this old line on my `.bashrc`:

```
# User scripts and python (pip) stuff
export PATH="$PATH:$HOME/.local/bin"
```

The comment reveals that I didn't know the full extent of what I was doing, but had set up `~/.local/bin` initially to install stuff through `pip` :smile: So it hadn't been default.

**Comment by Tweekism** [comment]
That's interesting haha.

I'm in Windows at the moment (downloading Warzone updates 🙄). When I'm back in Linux I'll have a look at my .rc file.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:30]: I'm not sure about this section, isn't it going to blow up a lot if we include every single package manager? Additionally (please let me know if this is too un-inclusive) I would argue that someone who can't figure out how to install `make`, `yarn` and `zip` on their system will have trouble making a valuable contribution to the code anyways.

So I would suggest shortening this to just mention the three dependencies. Same with the part on installing Node below; I would link to `nvm`'s repo/website but not include this level of detail. Also considering the fact that we would have to maintain/update this.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:88]: Not sure if this is obvious but once you have `configure`d, you can just run only `make install` and won't need to explicitly call build before since the Makefile will handle building for your system automatically. Maybe we could add that here.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:90]:
```suggestion
Once installed you can launch **viv** by calling `viv <some-file-or-directory>`
```

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:99]: This is not quite accurate: You don't need to build Vivify to run it in development mode (just having `yarn` installed, cloning the repo, and running `yarn` one time is enough), and if you installed it using pre-built binaries you can't run development mode at all.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:125]: This changed recently: `yarn viv` doesn't use the `viv` executable at all anymore, instead it just calls the server that now handles (almost) all CLI and options. This initially changed in https://github.com/jannis-baum/Vivify/pull/90 and further updates were made in https://github.com/jannis-baum/Vivify/pull/112 in case you want details

**Review by jannis-baum — CHANGES_REQUESTED** [review]
Hello! Thanks a lot for putting in the effort of writing such nice instructions here! I appreciate it a lot. See below for comments

File: docs/CONTRIBUTING.md
Comment (Tweekism) [reply docs/CONTRIBUTING.md:30]: Ok I'll revert this to a simple list of deps. I just noticed that it is wrong anyway, because the package on Fedora is called `yarkpkg` not `yarn`

A little anecdote, as I said earlier I took some inspiration from the Ladybird browser project. They have instructions for like 10 different environments as they also don't have any distro packages as it is early development, you have to build it. A couple of popular Linux YouTubers made  videos about it and so they had a flood of non-programmer types coming in to try out this "new browser", which justified the detailed instructions. We probably don't have to worry about that for the foreseeable future. Their issue tracker was fun for a few days there :P

File: docs/CONTRIBUTING.md
Comment (Tweekism) [reply docs/CONTRIBUTING.md:125]: I noticed something changed, but I didn't understand what was going on. I thought I must have broke something. I'll go look these up for context.

The other comments you raise are just me being stupid too late at night, I'll fix them up.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [reply docs/CONTRIBUTING.md:30]: Haha, interesting! I think we are fine with this since building is really not required. Either it's packaged for your system, or you download the prebuilt binaries from the latest release :)

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [reply docs/CONTRIBUTING.md:125]: Yes, no worries at all! The architecture of Vivify is probably a bit complicated to understand by just looking at the code because of the lazy starting and terminating of the server and all that. That's why I've been wanting to write a doc for that as well in #119.

Comment (Tweekism) [reply docs/CONTRIBUTING.md:125]: Let me know if I can help with that, obviously first I'll also need to understand it better.

Even if you throw up bullet points type rough draft, I can help flesh it out and prettify it. :)

Comment (jannis-baum) [reply docs/CONTRIBUTING.md]: Nice! This is actually probably a great way of doing it because if I write it entirely myself it might end up not being detailed enough to understand for new people. Will definitely get back to you on this one :)

**Comment by Tweekism** [comment]
Ok I've tightened it up significantly, and re-arranged a bit.

I'm going over it again double checking, let me know if thats better or if I missed anything.

**Review by jannis-baum — APPROVED** [review]
Nice, looks good! Let me know if everything is done from your side as well, if so I will merge this.

**Comment by Tweekism** [comment]
I just marked 2 backticks that I missed, other than that I'm happy.

**Comment by jannis-baum** [comment]
> I just marked 2 backticks that I missed, other than that I'm happy.

Oh, did you fix that? I don't see any new changes.

**Comment by Tweekism** [comment]
I marked it in the code review thing but didn't commit it. Maybe you can't see them. Want me to just push a new commit?

**Comment by Tweekism** [comment]
I don't know how to do things on my phone. I have my laptop now give me 2 minutes and I'll fix up these to tiny things

**Comment by Tweekism** [comment]
Done. 

I can squash those two new tiny commits into the previous one if you like.

File: docs/CONTRIBUTING.md
Comment (jannis-baum) [inline docs/CONTRIBUTING.md:39]: Let's also combine these into 1 line now that there's nothing left in between :) Can all happen in that squashed commit so that the total commits here is 2 ^^

**Review by jannis-baum — COMMENTED** [review]
> Done.
> 
> I can squash those two new tiny commits into the previous one if you like.

Yes sounds good!

**Comment by jannis-baum** [comment]
Nice! Thanks a lot again and congrats on your first official contributions to Vivify🥳

**Comment by Tweekism** [comment]
I'd like to see this project go far because you guys are so nice. I've seen so many small projects die because they were run by one dude who didn't really like to talk to people 😅

**Pull request merged**
